addon/helpers/optional.js

```diff
@@ -1,8 +1,8 @@
 import { helper } from 'ember-helper';
 
-export function optional([action]) {
+export function optional([action, ...args]) {
   if (typeof action === 'function') {
-    return action;
+    return action.bind(this, ...args);
```

Comment:
Yeah, that might be it.
But you might need to also factor in the arguments passed into the anonymous function you're returning from the helper.